third ludo review of group 4 by group 2 and 7

skjohansen opened the pull request.

aohzork committed:
Driver@aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver@aohzork

samirehsani19 committed:
Driver @samirehsani19

Har gjort färdig databas delen.
man kan inserta data
läsa data
ta bort data
ladda spel
hitta bästa spelare
redigera data

aohzork committed:
Driver@aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

samirehsani19 committed:
Driver @samirehsani19

Solved merge conflict

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

PontusGillenang committed:
Driver @pontusgillenang

Some test cleaning

PontusGillenang committed:
Revert "Driver @pontusgillenang"

This reverts commit 476dd9b5e5281be8f097ea433e5a5068eff4c50f.

skjohansen committed:
Classroom fixes

Adgnascor commented on Source/LudoGame/Program.cs:
```diff
@@ -1,19 +1,94 @@
+            Console.ReadKey();  
         }
     }
 }
```
Snyggt att ni har påbörjat arbetet av separationen mellan logik och GUI! Keep up the good work 💯 🥇

Adgnascor commented on Source/LudoGame/Menu.cs:
```diff
@@ -24,7 +24,8 @@ public static void Display()
 
-            int selected = MenuOptions(options);
+            //int selected = MenuOptions(options);
+            int selected = CreateInteractable.OptionMenu(false, options);
```
Bra namning, tydligare vad den innebär

Adgnascor commented on Source/LudoGameEngine/Dice.cs:
```diff
@@ -6,10 +6,21 @@ namespace LudoGameEngine
+
+        public Dice()
+        {
+            this.random = new Random();
+        }
+
+        public Dice(Random rnd)
+        {
+            this.random = rnd;
+        }
+
         public int Roll()
         {
-            Random rnd = new Random();
-            return rnd.Next(1, 7);
+            return random.Next(1, 7);
         }
```
👍

Adgnascor commented on Source/LudoGameEngine/Migrations/20200405105035_UpdatePK.cs:
```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace LudoGameEngine.Migrations
+{
+    public partial class UpdatePK : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+
+        }
+
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+
+        }
+    }
```
Känns som att dessa kanske inte skall vara kvar i projektet?

aohzork committed:
Driver @aohzork

Adgnascor reviewed (commented):
Bra jobbat boys! Och bra med tester, inte så mycket mer att tillägga eftersom vi gav en review nyligen!

Irvper77 reviewed (commented):
snyggt med tester 👍

SamBjork commented on Source/LudoGameEngine/CreateInteractable.cs:
```diff
@@ -0,0 +1,110 @@
+using System.Collections.Generic;
+using System.Text;
+
+namespace LudoGameEngine
```
Snygg meny-logik (som det ser ut att vara) men bör det verkligen ligga i GameEngine? Finns det någon speciell anledning, annars är det väl bra att ha i menyn eller Game projektet?

aohzork replied:
Kan nog vara både och faktiskt. Huvudsyftet med CreateInteractable är att hjälpa GameEngine. Ungefär som om  WPF är GameEngine, så är knappar du kan sätta ut visuellt tillhörande WPF, således GameEngine :)

SamBjork commented on Source/LudoGameEngine/Data/DataContext.cs:
```diff
@@ -21,5 +23,30 @@ protected override void OnConfiguring(DbContextOptionsBuilder builder)
+
+        protected override void  OnModelCreating(ModelBuilder Builder)
+        {
+            Builder.Entity<PlayerSession>().HasKey(c => new { c.SessionId, c.PlayerId });
```
Snyggt med modelbuilder mappning! :D

SamBjork commented on Source/LudoGameEngine/Data/GameData.cs:
```diff
@@ -0,0 +1,379 @@
+
+namespace LudoGameEngine.Data
+{
+    public class GameData
```
Riktigt bra strukturerat med denna klassen! 👍 Kanske är lite lång men förståeligt att vilja samla all databas logik här.

SamBjork commented on Source/LudoGameEngine/GameSession.cs:
```diff
@@ -37,6 +37,11 @@ public enum Color
 
+        public bool IsPlayerNameValid()
+        {
+            return !string.IsNullOrWhiteSpace(PlayerName);
```
Snygg felhantering!

SamBjork reviewed (commented):
Jättebra kod, ni verkar ha koll på det mesta! :) 👍

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

felhanterat och förbjudit samma session namn, och obehöriga vinnare till en session

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

samirehsani19 committed:
Driver @samirehsani19

checked everything works.

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

lots of refractoring & worked alot on design

PontusGillenang committed:
Driver @pontusgillenang

PontusGillenang committed:
Driver @pontusgillenang

PontusGillenang committed:
Revert "Driver @pontusgillenang"

This reverts commit 65ae0587300ea49b1af3c1bee0e771b7ed5a4ef8.

PontusGillenang committed:
Driver @pontusgillenang

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

lots of fixes, refactoring code and implement changes & features

aohzork committed:
Driver @aohzork

minor adjustments

samirehsani19 committed:
Driver @samirehsani19

contected some of database class and methods to menu

PontusGillenang committed:
Driver @pontusgillenang

PontusGillenang committed:
Driver @pontusgillenang

PontusGillenang committed:
Driver @pontusgillenang

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

samirehsani19 committed:
driver @samirehsani19

samirehsani19 committed:
Driver @samirehsani19

aohzork committed:
Driver @aohzork

PontusGillenang committed:
Driver @pontusgillenang

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

Created method for session names

aohzork committed:
Driver @aohzork

aohzork committed:
Driver@ aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

samirehsani19 committed:
Driver @samirehsani19

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

can move pieces to goal, current global position and new global position

aohzork committed:
Driver @aohzork

different colors on common board move around independently

aohzork committed:
Driver @aohzor

Load Game works from creating game with pieces position 0, from preinserted db-content

aohzork committed:
Driver@aohzork

ingame save and save winner methods prepared,  making some code more readable.

aohzork committed:
Driver @aohzork

implemented winner and activate save game functions

aohzork committed:
Driver @aohzork

small fixes in gamesession

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

PontusGillenang committed:
Driver @pontusgillenang

samirehsani19 committed:
Driver @samirehsani19

aohzork committed:
Driver @aohzork

PontusGillenang committed:
Revert "Driver @pontusgillenang"

This reverts commit df37b1bacdd74cfba2dc54ba86f7cd9f3f476284.

aohzork committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

simplfied movementmethods in earlier push, and catalogueizing some  in this.

aohzork committed:
Driver @aohzork

aohzork committed:
Driver@aohzork

aohzork committed:
Driver @aohzork

knockout other player

aohzork committed:
Driver @aohzork

move behind some more work

aohzork committed:
Driver @aohzork

working move behind & knockout + win

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

aohzork committed:
Driver @aohzork

aohzork committed:
Driver@aohzork

samirehsani19 committed:
Driver @samirehsani19

solved dubblicate names and position of piece

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

rättat till koden i loopen

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @ohzork

updating documentation and cleaning som code

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

cleanup

PontusGillenang committed:
Driver @pontusgillenang

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

Snyggat till utskriften i game load

aohzork committed:
Driver @aohzork

refactoring & cleaning up

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

Solved a problem with saving the winner and solved other small problems

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

some fixes

aohzork committed:
Driver @aohzork

PontusGillenang committed:
Driver @pontusgillenang

PontusGillenang committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

uppdatering av klassdiagram

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

uppdaterar klassinformation

aohzork committed:
Driver @ aohzork

samirehsani19 committed:
Driver @samirehsani19

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

samirehsani19 committed:
Driver @samirehsani19

solved problem with position updating

samirehsani19 committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-group-4

aohzork committed:
Driver @aohzork

aohzork committed:
Driver @aohzork

updating klassbeskrivning & separation of code

samirehsani19 committed:
Driver @samirehsani19

Created wining song

samirehsani19 committed:
Driver @samirehsnai19

aohzork committed:
Driver @aohzork
